src/views/v3/Bridge/AssetPicker/PickerBottomSheet.tsx

```diff
@@ -0,0 +1,160 @@
+import React, { useMemo } from 'react';
+import { useTheme } from '@mui/material';
+import Box from '@mui/material/Box';
+import Stack from '@mui/material/Stack';
+import SwipeableDrawer from '@mui/material/SwipeableDrawer';
+import { type Chain } from '@wormhole-foundation/sdk';
+
+import type { ChainConfig } from 'config/types';
+import type { WalletData } from 'store/wallet';
+import { Token } from 'config/tokens';
+import { Balances } from 'utils/wallet/types';
+import ChainList from 'views/v3/Bridge/AssetPicker/ChainList';
+import TokenList from 'views/v3/Bridge/AssetPicker/TokenList';
+
+interface AssetPickerDrawerProps {
+  isDrawerOpen: boolean;
+  setIsDrawerOpen: (value: boolean) => void;
+  chainList: Array<ChainConfig>;
+  chainConfig?: ChainConfig;
+  showChainSearch: boolean;
+  setShowChainSearch: (value: boolean) => void;
+  wallet: WalletData;
+  sortedTokens: Token[];
+  balances: Balances;
+  isFetchingBalances: boolean;
+  isConnectingWallet?: boolean;
+  isFetchingTokens?: boolean;
+  token?: Token;
+  sourceToken?: Token;
+  isSource: boolean;
+  searchQuery: string;
+  setSearchQuery: (value: string) => void;
+  onChainSelect: (value: Chain) => void;
+  onTokenSelect: (value: Token) => void;
+}
+
+function AssetPickerDrawer({
+  isDrawerOpen,
+  setIsDrawerOpen,
+  chainList,
+  chainConfig,
+  showChainSearch,
+  setShowChainSearch,
+  wallet,
+  sortedTokens,
+  balances,
+  isFetchingBalances,
+  isConnectingWallet,
+  isFetchingTokens,
+  token,
+  sourceToken,
+  isSource,
+  searchQuery,
+  setSearchQuery,
+  onChainSelect,
+  onTokenSelect,
+}: AssetPickerDrawerProps) {
+  const theme = useTheme();
+
+  // Drawer handle - static content, no need to memoize
+  const drawerHandle = (
+    <Stack alignItems="center" paddingBottom="4px" paddingTop="8px">
+      <Box
+        sx={{
+          width: '40px',
+          height: '5px',
+          backgroundColor: theme.palette.text.secondary,
+          borderRadius: '8px',
+        }}
+      />
+    </Stack>
+  );
+
+  // Chain list - memoize because it involves chain selection logic
+  const chainListContent = useMemo(
+    () => (
+      <ChainList
+        chainList={chainList}
+        selectedChainConfig={chainConfig}
+        showSearch={showChainSearch}
+        setShowSearch={setShowChainSearch}
+        wallet={wallet}
+        onChainSelect={onChainSelect}
+      />
+    ),
+    [
+      chainList,
```

Comment:
Also note that memo only works over shallow comparison. So it works well over simple primitive values, but can fail if more complex objects are themselves are not memoized. You can probably test how many times a console.log happens to see if it is behaving as needed.